config/initializers/rack_attack.rb

```diff
@@ -0,0 +1,23 @@
+# Safelist requests from signed in users in the web app
+Rack::Attack.safelist(
+  'mark authenticated request from web app users as safe'
+) do |request|
+  request.env['HTTP_COOKIE'].present? &&
+    request.env['HTTP_AUTHORIZATION'].blank?
+end
+
+# Safelist GET requests
+Rack::Attack.safelist('mark get requests as safe') { |request| request.get? }
+
+# Throttle GraphQL API requests to a configurable value per second that defaults to 20 requests/second
+Rack::Attack.throttle(
+  'limits GraphQL api requests using API token',
+  limit: ENV['GRAPH_API_RATE_LIMIT']&.to_i || 20,
+  period: 1
+) do |request|
+  header = request.env['HTTP_AUTHORIZATION']&.strip
+  api_token = header.split(' ')[-1] if header.present?
+  if request.path == '/graphql' && request.post? && api_token.present?
+    Digest::SHA2.base64digest(api_token)
+  end
```

Comment:
This block appears to return either a digest string, or `nil`. I've gone through docs of `rack-attack`, and I can't figure out what this gem's behaviour is if the returned _key_ is `nil`. Does it not throttle the request?

This needs to be tested.

Reply:
It ignores and does not attempt to throttle such requests: https://github.com/rack/rack-attack/blob/933c0576b8ab1b398d627f87aef24585fe263135/lib/rack/attack/throttle.rb#L27